### PR TITLE
feat(dashboard): use react-components RE components to update dashboard RE for modeled and unmodeled data

### PR DESCRIPTION
**Pull request by ssjagad**
## Overview
- updated RE to use new RE components in react-components package for modeled and unmodeled tab
- hooked up new RE footer to each tab to update query accordingly
- perform correct disabling/selection of buttons and selection states based on amount of widgets selected + selected widget type
- added prop to allow for descriptions to be added for explorer table headers and added it to the modeled RE component to match current RE implementation
- added ariaLabel to each row of table to allow UI testing helpers to remain the same
- edited MSW to work with new RE components in listAssociatedTimeSeriesCall
- removed 2 kpi tests regarding multiselect since selection for KPI properties is now singular through new components

## Verifying Changes

https://github.com/awslabs/iot-app-kit/assets/145582655/536f5405-7106-4fe8-9989-bdead370d9f7

Screenshot 2024-07-11 at 8 42 27 AM

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/constants.ts [hunk 1]
```diff
@@ -1,5 +1,83 @@
...
 export const STICKY_BUTTON_WIDTH_FACTOR = 25;
+
+const DEFAULT_STRING_FILTER_OPERATORS = [
+  '=',
+  '!=',
+  ':',
+  '!:',
+] as const satisfies readonly ResourceFieldFilterOperator[];
+
+export const DEFAULT_TIME_SERIES_TABLE_DEFINITION: TableResourceDefinition<TimeSeriesResourceWithLatestValue> =
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/constants.ts:20]: Why are these duplicated?

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 2]
```diff
@@ -1,33 +1,83 @@
...
+  AssetExplorerProps,
+  AssetPropertyExplorer,
+  AssetPropertyExplorerProps,
+  TimeSeriesExplorer,
+  TimeSeriesExplorerProps,
+} from '@iot-app-kit/react-components';
+import { IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { ResourceExplorerFooter } from './footer/footer';
+import { getPlugin } from '@iot-app-kit/core/dist/es/plugins/pluginsRegistry';
+import { Box } from '@cloudscape-design/components';
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:26]: Make sure you're importing components from Cloudscape like:

```ts
import Box from '@cloudscape-design/components/box';
```

The way you are importing is not tree-shakeable.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 3]
```diff
@@ -1,33 +1,83 @@
...
 import { useQuery } from '../useQuery';
 import { AssetModelDataStreamExplorer } from './assetModelDataStreamExplorer/assetModelDataStreamExplorer';
+import {
+  AssetExplorer,
+  AssetExplorerProps,
+  AssetPropertyExplorer,
+  AssetPropertyExplorerProps,
+  TimeSeriesExplorer,
+  TimeSeriesExplorerProps,
+} from '@iot-app-kit/react-components';
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:22]: Nit: Use type imports for types.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 4]
```diff
@@ -1,33 +1,83 @@
...
+  useEffect(() => {
+    setSelectedAssetProperties([]);
+  }, [selectedAssets]);
+
+  useEffect(() => {
+    setSelectedAssetProperties(
+      selectedAssetProperties.filter(
+        (item) =>
+          !isModeledPropertyInvalid(
+            item.dataType as PropertyDataType,
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:74]: Why does this need to be cast?

File: packages/dashboard/src/msw/iot-sitewise/resources/assets/index.ts [hunk 5]
```diff
@@ -44,22 +44,13 @@ class AssetHierarchyClient {
...
-    const hierarchy = node?.asset.assetHierarchies?.find(
-      ({ id }) => id === hierarchyId
-    );
-    // childAssetModelId is not on Assets - We keep it in the translation from AssetModel to Asset to enable this search.
-    const hierarchyAsAssetModelHierarchy = hierarchy as NonNullable<
-      AssetModel['assetModelHierarchies']
-    >[number];
-    const childAssetModelId = hierarchyAsAssetModelHierarchy.childAssetModelId;
+
+    console.log(node);
```
Comment (tracy-french) [inline packages/dashboard/src/msw/iot-sitewise/resources/assets/index.ts:50]: Nit: Remove console.log.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 6]
```diff
@@ -54,33 +104,141 @@ export function IoTSiteWiseQueryEditor({
...
+          }}
+          tableResourceDefinition={DEFAULT_TIME_SERIES_TABLE_DEFINITION}
+        />
+        <ResourceExplorerFooter
+          addDisabled={
+            selectedTimeSeries.length === 0 || selectedWidgets.length === 0
+          }
+          onReset={() => setSelectedAssetProperties([])}
+          onAdd={() => {
+            console.log(selectedTimeSeries);
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:222]: Nit: Remove console.log.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 7]
```diff
@@ -54,33 +104,141 @@ export function IoTSiteWiseQueryEditor({
...
+            handleClickAddUnmodeledDataStreams(
+              selectedTimeSeries as unknown as UnmodeledDataStream[]
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:224]: Why does this need to be cast like this?

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx [hunk 8]
```diff
@@ -54,33 +104,141 @@ export function IoTSiteWiseQueryEditor({
...
+                : `Deselect asset ${asset.name}`,
+          }}
+        />
+        {selectedAssets.length > 0 && (
+          <AssetPropertyExplorer
+            requestFns={iotSiteWise}
+            parameters={selectedAssets}
+            selectionMode={
+              selectedWidgets.at(0)?.type === 'kpi' ? 'single' : 'multi'
+            }
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/iotSiteWiseQueryEditor.tsx:159]: I would pull this handling out of the render to make it more obvious.

File: packages/react-components/src/components/resource-explorers/explorers/asset-explorer/types.ts [hunk 9]
```diff
@@ -22,6 +23,8 @@ export interface AssetExplorerProps
   onSelectAsset?: OnSelectResource<AssetResource>;
   selectedAssets?: SelectedResources<AssetResource>;
   isAssetDisabled?: IsResourceDisabled<AssetResource>;
+  description?: string;
```
Comment (tracy-french) [inline packages/react-components/src/components/resource-explorers/explorers/asset-explorer/types.ts:26]: Do we need a test to validate the handling of this field?

File: packages/dashboard/src/components/dashboard/index.test.tsx
Comment (tracy-french) [inline packages/dashboard/src/components/dashboard/index.test.tsx:1]: Do we have tests on the dashboard which validate the queries are created correctly? Also, are there tests to capture the change to no longer require a workspace to be selected to perform search?

File: packages/dashboard/src/components/queryEditor/queryEditor.tsx
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/queryEditor.tsx:1]: We should delete the parts of the query editor which are no longer used.

Comment (ssjagad) [reply packages/dashboard/src/components/queryEditor/queryEditor.tsx:1]: Gonna do that in follow up PR. This PR is just getting the modeled and unmodeled tabs to use the new RE. Unless you think we should include it in this one? Might pollute the actual changes if anyone wants to look..

**Pull request closed without merging**